### PR TITLE
Remove references to PaaS

### PR DESCRIPTION
**Pull request by cadmiumcat**
### What problem does this pull request solve?

Trello card: https://trello.com/c/DIr6jt2R/177-remove-paas-references-in-code-base
Removing references to PaaS from our code.

### Things to consider when reviewing

This is my naive attempt at doing it. I don't know if it breaks anything and also not 100% how to check if it does

- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 class ApplicationController < ActionController::Base
   include Pundit::Authorization
+  delegate :request_id, to: :request
```
Comment (danworth) [inline app/controllers/application_controller.rb:5]: hmm rebased and aligned with main but I'm not sure what this change is, is it unintentional?

Comment (cadmiumcat) [reply app/controllers/application_controller.rb:5]: This is to replace [`def request_id`](https://github.com/alphagov/forms-admin/pull/588/files#diff-766c34fd6533171eaf54300c153f89d6002c35c02cfc9c5b219251f85180ad07L67-L76)
```
def request_id
    request.request_id
end
```
Rubocop told me to do it 👀 

But I am not 100% that it is the right way forward

Comment (lfdebrux) [reply app/controllers/application_controller.rb:5]: What was the cop ID?

Comment (lfdebrux) [reply app/controllers/application_controller.rb:5]: Further question, why do we need to define `request_id` at all?

Comment (DavidBiddle) [reply app/controllers/application_controller.rb:5]: Looks like [Rails/Delegate](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsdelegate)

Comment (cadmiumcat) [reply app/controllers/application_controller.rb:5]: @lfdebrux I was wondering too whether it needed to be defined at all - I wasn't sure if it was a Ruby thing.

Comment (cadmiumcat) [reply app/controllers/application_controller.rb:5]: I will push a temporary change without this line and see if it breaks anything 😁

Comment (cadmiumcat) [reply app/controllers/application_controller.rb]: ohh @lfdebrux, did you mean we could remove all references to `request_id`?

Comment (lfdebrux) [reply app/controllers/application_controller.rb]: I'm just wondering why we need to define `request_id` on the `ApplicationController` class. As far as I can see in our code, the only place where we use the `request_id` method is in `set_request_id`:

https://github.com/alphagov/forms-admin/blob/10fcf54a1bf3878ca5f5208629eacd21d81f9f6c/app/controllers/application_controller.rb#L56C33-L56C33

If we're setting `request.request_id` to itself that seems kind of redundant. So we should be able to get rid of that line also, right? Then what's the point of the `request_id` method?

Comment (danworth) [reply app/controllers/application_controller.rb]: Yeah I was thinking the same Laurence it seems unnecessary in the context of ApplicationController, but then, does any controller inheriting from ApplicationController (or somewhere else) use it, I didn't have time to go digging/grepping. Just reminded of the whole `@current_user` vs `current_user` etc...

Comment (danworth) [reply app/controllers/application_controller.rb]: @cadmiumcat 
>I will push a temporary change without this line and see if it breaks anything 😁

If you want to try removing it you'll also need to remove the reference to the `request_id` method it was providing in  [set_request_id](https://github.com/alphagov/forms-admin/pull/588/files#diff-766c34fd6533171eaf54300c153f89d6002c35c02cfc9c5b219251f85180ad07R54) which is why the tests are currently failing: e.g https://github.com/alphagov/forms-admin/actions/runs/5819019255/job/15776594974?pr=588#step:9:3213

Comment (cadmiumcat) [reply app/controllers/application_controller.rb]: Got it, brain has caught up now. I've removed both things, thanks all 🙌

**Review by alice-carr — APPROVED** [review]

**Pull request merged**